### PR TITLE
Feature/updated restructure 2

### PR DESCRIPTION
**Pull request by amk0**
Merged all the latest changes master into the restructure branch.
Added permission support in the actions
Updated remove recursive

### CODE REVIEW

File: Module.php [hunk 1]
```diff
@@ -102,7 +102,7 @@ class Module extends \yii\base\Module
      * Active / deactivate the filesystem recursive folder delete
      * @var bool
      */
-    public $deleteRecursive = false;
+    public $deleteRecursive = true;
```
Comment (schmunk42) [inline Module.php:105]: Please do not change the default, can be configured in project config.

**Review by schmunk42 — CHANGES_REQUESTED** [review]
@eluhr Didn't you refactor the code into single compnents several months ago, which would now be reverted?

**Comment by handcode** [comment]
@schmunk42 this branch is based on the feature/restructure branch from @eluhr which was never merged #13

**Comment by schmunk42** [comment]
Thank you!

**Pull request merged**